File: DCAnalyticsWebShell/ClientApp/src/models/Configuration.ts
import { Questionaire } from "./Questionaire";
import { EnumList } from "./EnumList";
import { User } from "./User";
import { UserRight } from "./UserRight";
import { DbInfo } from "./DbInfo";
import { FieldInspection } from "./FieldInspection";
import { Training } from "./Training";
import { Certification } from "./Certification";
import { Purchase } from "./Purchase";
import { Settings } from "./Settings";
import { SyncDirection, ConfigurationTypes } from "../enums/enums";
import { ConfigurationVersion } from "./ConfigurationVersion";
import { Client } from "./Client";

export class Configuration {
  public OID: number;
  public Key: string;
  public FileName: string;
  public Icon: string;
  public SyncDirection: SyncDirection;
  public CreatedOn: Date;
  public ModifiedOn: Date;
  public ModifiedBy: string;
  public RequireLocation: boolean;
  public Name: string;
  public Type: ConfigurationTypes
  public Domain: string;
  public Database: string;
  public Description: string;
  public ConfigVersion : ConfigurationVersion;
  public Version: string;
  public Status: number;
  public Questionaires: Questionaire[];
  public EnumerationLists: EnumList[];
  public Users: User[];
  public UserRights: UserRight[];
  public DbInfo: DbInfo;
  public Settings: Settings;
  public Purchases: Purchase;
  public Certifications: Certification[];
  public Trainings: Training[];
  public Regions: EnumList;
  public Products: EnumList;
  public Prices: EnumList;
  public Inspections: FieldInspection[];
  public Client : Client;
  public Deleted: boolean
}

File: DCAnalyticsMobile/DCAnalyticsMobile/Models/Configuration.cs
using DCAnalytics;
using System;
using System.Collections.Generic;
using System.Text;

namespace DCAnalyticsMobile.Models
{
    public class Configuration: DataCollectionObject
    {
        public string FileName { get; set; }
        public string Icon { get; set; }
        public SyncDirection SyncDirection { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime ModifiedOn { get; set; }
        public string ModifiedBy { get; set; }
        public bool RequireLocation { get; set; }
        public string Domain { get; set; }
        public string Database { get; set; }
        public string Description { get; set; }
        public string Version { get; set; }
        public int Status { get; set; }
        public List<Questionaire> Questionaires { get; set; }
        public List<EnumList> EnumerationLists { get; set; }
        public List<User> Users { get; set; }
        public List<UserRight> UserRights { get; set; }
        public DbInfo DbInfo { get; set; }
        public Settings Settings { get; set; }
        public List<Purchase> Purchases { get; set; }
        public List<Certification> Certifications { get; set; }
        public List<Training> Trainings { get; set; }
        public EnumList Regions { get; set; }
        public EnumList Products { get; set; }
        public EnumList Prices { get; set; }
        public List<FieldInspection> Inspections { get; set; }
        public int Sequence { get; set; }
    }
}
